signing_service/README.md

```diff
@@ -0,0 +1,68 @@
+# signing-service
+
+Stand-alone Python application used for signing Ethereum transactions received via MiddleMan protocol.
+
+
+### Install
+
+You can install this library:
+
+``` bash
+python setup.py install
+```
+
+
+### Install for development
+
+You can install this library for development:
+
+``` bash
+python setup.py develop
+```
+
+
+### Running tests
+
+You can run automated tests with:
+
+``` bash
+python -m pytest -p no:django .
+```
+
+
+### Running options
+
+positional arguments:
+  concent_cluster_host  Host or IP address of a service on Concent cluster, to
```

Comment:
Please don't hard-code this here. No one will remember to update it when it changes in the code so it'll be always out of date.

Instead, please only describe parameters that need special attention. Like `--sentry-environment`. And just say in general what is optional and what is needed to run it (which keys, where the user is expected to get them from, etc). Basically the information that is not normally included in the `--help` output.

Reply:
> Please don't hard-code this here. No one will remember to update it when it changes in the code so it'll be always out of date.

I don't agree with this assumption. It sounds like: no matter what, we will do it wrong anyway. If we pay attention to this during reviews, it won't get outdated.

I have rephrased this part a little bit, but I think mentioning all parameters is ok. There are not so many of them, and I couldn't find a pattern in which some are more important than other.

Reply:
> I don't agree with this assumption. It sounds like: no matter what, we will do it wrong anyway.

Sorry, I didn't want to come off as offensive. I don't mean that they'll go out of sync due to someone's incompetence or bad will. I simply accept that if you have two identical pieces of code/text and nothing to enforce that they're identical, they'll go out of sync sooner or later. It just happens.

Having to pay attention doing reviews is fine but the best strategy is to simply avoid duplication if it's not necessary. I think that in this case the duplication is not worth it because you can easily get this information from `--help`.

Also, I assumed that you just did copy&paste of the output of the `--help` option. If you wrote it from scratch and it does not contain exactly the same information, that's a different story. I did not compare them so I don't know.

> I have rephrased this part a little bit, but I think mentioning all parameters is ok. There are not so many of them, and I couldn't find a pattern in which some are more important than other.

I would not include all the parameters here. I think a high level description is better because it does not rely as much on the implementation details:

```
To run Signing Service you need the following:
- Address and port of a Concent cluster to connect to.
- A private key for signing sent messages.
- Concent's public key for verifying signatures of received messages.
- [Sentry DNS](https://docs.sentry.io/quickstart/#configure-the-dsn) if you want to submit error reports to a Sentry instance (optional but recommended).
    - If you want to submit reports to the instance used by Concent (recommended) please also use `--sentry-environment` option to set the `environment` tag to `concent-staging`, `concent-testnet` or `concent-mainnet`, depending on which cluster you're connecting to. If you're submitting to any other instance, you can set this tag to anything you want or even simply omit it. Its only purpose is to let us discern reports from different environments in Sentry.
```

That's just a draft. Option names and explaining how we pass secrets should probably be added too.

I wouldn't really mention the reconnect delay unless there anything more to say than `--help` already does. You might just mention that there's more options and to use `--help` to see them all.

Anyway, that's my opinion. But the `README` update was meant to be just a small addition to this pull request. I don't want to hold the whole pull request back so if you still strongly disagree, please remove this section of the `README`, mention only `--sentry-environment` (because that's what this pull request is about) and let's move the discussion and the removed changes to #704.

Reply:
Looks fine now.